### PR TITLE
refactor: RHACM backend plugin

### PR DESCRIPTION
**Pull request by tumido**
Refactor RHACM backend in preparation for future backend work:

- rename `clusterStatus` settings to `rhacm`
- break up the unnecessary `StatusCheck` class that was instatiated only once and had each public method called once only. Broken into API helper functions and constants
- remove `checkClusterNames` method completely since obtaining cluster status does not depend on the cluster being available in `kubernetes` plugin. It's unrelated (in future can be raised as a warning on the entity side).

### CODE REVIEW

**Review by SamoKopecky — COMMENTED** [review]
Looks really clean! Just have one small suggestion.

File: plugins/rhacm-backend/src/helpers/config.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { Config } from '@backstage/config';
+import { HUB_CLUSTER_CONFIG_PATH } from '../constants';
+
+const CLUSTERS_PATH = 'kubernetes.clusterLocatorMethods';
```
[the remark this answers is not in the document]
Comment (tumido) [reply plugins/rhacm-backend/src/helpers/config.ts:4]: I don't think we should export it outside of the `config` helper context. I don't think this constant has any place to be reused elsewhere...

[the remark this answers is not in the document]
Comment (tumido) [reply plugins/rhacm-backend/src/helpers/config.ts:4]: Hm.. but the same logic applies to `HUB_CLUSTER_CONFIG_PATH`... I'll move that one in here as well.

**Pull request merged**